src/autonlp/project.py

```diff
@@ -262,11 +253,59 @@ def upload(self, filepaths: List[str], split: str, col_mapping: Dict[str, str]):
             http_post(path=f"/projects/{self.proj_id}/data/add", payload=payload, token=self._token)
             logger.info(f"[{idx + 1}/{len(filepaths)}] ✅ Success!")
 
-    def train(self):
+    def train(self, noprompt=False):
         """Starts training on the models"""
+        logger.info("🔎 Running an estimate of the training cost...")
+        dataset_repo = self._clone_dataset_repo()
+        local_dataset_dir = dataset_repo.local_dir
+        total_number_of_lines = 0
+
+        for root, dirs, files in os.walk(os.path.join(local_dataset_dir, "raw")):
+            for file_path in files:
+                with open(os.path.join(root, file_path), "r", encoding="utf-8", errors="ignore") as f:
+                    total_number_of_lines += sum(1 for line in f)
+        try:
+            payload = {"username": self.user, "language": self.language, "num_train_samples": total_number_of_lines}
+            cost_estimate = http_post(path="/zeus/estimate", token=self._token, payload=payload).json()
+        except requests.exceptions.HTTPError as err:
+            if err.response.status_code == 404:
+                raise ValueError("❌ Unable to estimate") from err
+            raise
+
+        print(
+            "\n"
+            "💰 The cost for training this project should range from:\n"
+            f"  {BOLD_TAG}USD {cost_estimate['cost_min']} to USD {cost_estimate['cost_max']}.{RESET_TAG}\n\n"
+            f"⚠ This is a {BOLD_TAG}rough estimate{RESET_TAG} based on the {BOLD_TAG}number of lines{RESET_TAG} in the files you uploaded.\n"
+            "  The actual cost of training can be lower or higher!\n"
```

Comment:
suggestion, provided we can force the invoice to be in estimated range:
from
            "💰 The cost for training this project should range from:\n"
            f"  {BOLD_TAG}USD {cost_estimate['cost_min']} to USD {cost_estimate['cost_max']}.{RESET_TAG}\n\n"
            f"⚠ This is a {BOLD_TAG}rough estimate{RESET_TAG} based on the {BOLD_TAG}number of lines{RESET_TAG} in the files you uploaded.\n"
            "  The actual cost of training can be lower or higher!\n"
to
            "💰 The training cost for this project will be in this range:\n"
            f"  {BOLD_TAG}USD {cost_estimate['cost_min']} to USD {cost_estimate['cost_max']}{RESET_TAG}\n\n"
            f"⚠ This {BOLD_TAG}estimated range{RESET_TAG} is based on the {BOLD_TAG}number of lines{RESET_TAG} in your uploaded training data.\n"
            "  Once training is complete, we will send you an email invoice for the actual training cost within that range.\n"

Reply:
different suggestion, if we keep the existing process (range is indicative, actual cost may vary):
from
            "💰 The cost for training this project should range from:\n"
            f"  {BOLD_TAG}USD {cost_estimate['cost_min']} to USD {cost_estimate['cost_max']}.{RESET_TAG}\n\n"
            f"⚠ This is a {BOLD_TAG}rough estimate{RESET_TAG} based on the {BOLD_TAG}number of lines{RESET_TAG} in the files you uploaded.\n"
            "  The actual cost of training can be lower or higher!\n"
to
            "💰 The training cost for this project is in this estimated range:\n"
            f"  {BOLD_TAG}USD {cost_estimate['cost_min']} to USD {cost_estimate['cost_max']}{RESET_TAG}\n\n"
            f"⚠ This range is an {BOLD_TAG}estimate{RESET_TAG} based on the {BOLD_TAG}number of lines{RESET_TAG} in your uploaded training data.\n"
            "  Once training is complete, we will send you an email invoice for the actual training cost.\n"

Reply:
Based on Slack thread, let's go with the first suggestion (we can force the invoice to be in the estimated range)
https://huggingface.slack.com/archives/C01QLFTRF62/p1615488753007300?thread_ts=1615462617.002500&cid=C01QLFTRF62